cmd/build.go

```
@@ -317,3 +334,37 @@ func createKubernetesExecutor(cfg kanikoConfig) (*kaniko.KubernetesExecutor, err
 
 	return executor, nil
 }
+
+func createGossTestRunner(opts buildOpts, workingDir string) *goss.TestRunner {
+	runnerOpts := goss.TestRunnerOptions{
+		ReportsDirectory: junitReportsDirectory,
+		WorkingDirectory: workingDir,
+		JUnitReports:     !opts.DisableJunitReports,
+	}
+
+	if opts.Goss.Executor.Kubernetes.Enabled && !opts.LocalOnly {
+		executor, err := createGossKubernetesExecutor(opts.Goss)
+		if err != nil {
+			logrus.Fatalf("Cannot create goss executor: %v", err)
```

You should return the error and handle it correctly in the parent function